libc/CMakeLists.txt

```diff
@@ -600,16 +600,19 @@ add_enclave_library(
   ${MUSLSRC}/search/lsearch.c
   ${MUSLSRC}/select/select.c
   ${MUSLSRC}/select/poll.c
-  ${MUSLSRC}/stat/mkdir.c
+  ${MUSLSRC}/search/hsearch.c
```

Comment:
The test is currently disabled: https://github.com/openenclave/openenclave/blob/82a698498910332eb637af80b542d2e061ee3a47/tests/libc/tests.cmake#L318

Reply:
Added test. Also fixed a bug with debugmalloc not setting errno that the test revealed.